### ISSUE TITLE
Conflict with "Share with SNS" add-on

### ISSUE THREAD

**Issue by asamuzaK** [piroor/treestyletab#2155]
## Short description
Under certain conditions, tab that should be focused become strange.

## Steps to reproduce

 1. Start Firefox with clean profile.
 2. Install TST.
 3. Install [Share with SNS](https://addons.mozilla.org/en-US/firefox/addon/share-with-sns/).
 4. Enable `Twitter` (or anything else) from Share with SNS options.
 5. Open https://example.com (Tab A) and https://example.net (Tab B).
 6. Click Tab A then right click on Tab A and select `Share Tab with Twitter`.
    New tab will be created (Tab C) as a child of Tab A.
 7. Make sure new Tab C is active (has focus).
    *Without moving focus from Tab C*, right click on Tab B and select `Share Tab with Twitter`.
    New tab will be created (Tab D) as a child of Tab B, and Tab C will get collapsed.

## Expected result
Tab D is active and content of Tab D shown in content area.
```
Tab A
\- Tab C (collapsed)
Tab B
\- Tab D (active) --- content of Tab D shown
```

## Actual result
Real tab:
Tab A is active and content of Tab A shown in content area.

TST:
Tab D is active but content of Tab A shown in content area.
Also, Tab A has focus (selected) too.
```
Tab A (focused) --- content of Tab A shown
\- Tab C (collapsed)
Tab B
\- Tab D (active)
```

## Environment

 * Platform (OS): Win 10
 * Version of Firefox:  67.0a1
 * Version (or revision) of Tree Style Tab:  2.7.18

Note:
Current Share with SNS has some codes for old TST `fake-context-menu`.
You can try pre-release which removed those codes [shareSNS/dest at master · asamuzaK/shareSNS](https://github.com/asamuzaK/shareSNS/tree/master/dest).
The result is the same.

**Comment by piroor** [comment]
It looks caused by this section: https://github.com/piroor/treestyletab/blob/92e3c7d87a4221aa48ada2a19fbe6b0f32c83936/webextensions/common/tree.js#L709
When the problem is reproduced:

 1. Share SNS opens a new tab "D".
 2. Share SNS moves focus to the tab "D".
 3. TST observes a `tabs.onCreated` event and TST collapses a tree of "A". The tab "C" is going to be collapsed.
 4. TST moves focus to the tab "A", because it is most nearest visible (not collapsed) tab.

This strange result is caused with parallelly produced multiple asynchronous operations. Hmm...

**Comment by asamuzaK** [comment]
You can check if created tab is active or not like this.
```
browser.tabs.onCreated.addListener(tab => {
  const {active} = tab;
  if (active) {
    // do something
  }
});
```

**Comment by piroor** [comment]
It looked not so easy because both TST's onActivated and onCreated handlers are asynchronous. After a research I thought that inserting onActivated operation to onCreated seemed not introducing side effects. Thus the commit 6003f9a should just solves this issue.

**Comment by asamuzaK** [comment]
Confirm fixed with TST 2.7.21
